specifyweb/frontend/js_src/lib/components/Toolbar/RecordSets.tsx

```diff
@@ -24,38 +24,53 @@ import { TableIcon } from '../Molecules/TableIcon';
 import { hasToolPermission } from '../Permissions/helpers';
 import { OverlayContext } from '../Router/Router';
 import { EditRecordSet } from './RecordSetEdit';
+import { usePaginator } from './Paginator';
 
 export function RecordSetsOverlay(): JSX.Element {
   const handleClose = React.useContext(OverlayContext);
+  const [currentPage, setCurrentPage] = React.useState<number>(0);
+  const [rowsPerPage, setRowPerPage] = React.useState<number>(10);
 
   const recordSetsPromise = React.useMemo(
     async () =>
       fetchCollection('RecordSet', {
         specifyUser: userInformation.id,
         type: 0,
-        limit: 5000,
+        limit: rowsPerPage,
         domainFilter: true,
         orderBy: '-timestampCreated',
```

Comment:
This is not right. What's happening here is that you are fetching unsorted data for a single page and then sorting it - this sorts the data within the page only. Instead, sorting should be done on the back-end and then the front-end would retrieve a slice of that data for a given page.

Refactor the code to pass sortConfig in orderBy, instead of sorting the record sets on the front-end